### PR TITLE
Dataset

### PR DESCRIPTION
**Pull request by andrevitorelli**

### CODE REVIEW

**Comment by EiffL** [comment]
I think it's looking good :-) but could you update your branch to `main`? so that I can see if it works when the package is installed?

File: autometacal/python/datasets/gal_gen.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+"""gal_gen dataset."""
+import tensorflow_datasets as tfds
+import tensorflow as tf
+import numpy as np
+import galsim
+from galaxies import generate_galaxy
```
Comment (EiffL) [inline autometacal/python/datasets/gal_gen.py:6]: This doesn't seem to work, maybe use this instead for now:
```python
from .galaxies import generate_galaxy
```

**Review by EiffL — CHANGES_REQUESTED** [review]
Looking good :-) Thanks Andre! 
I found an issue when trying to import the GalGen object, but that should be easily fixable.

**Comment by b-remy** [comment]
I added a `BuilderConfig` class to the `gal_gen.py` dataset file that enables to creates variants.

We can specify for instance stamp size, pixel size or the flux, parameters that are used to create galaxy images with `datasets/galaxies.py`.

Specifying the name and parameters of your variant such as:
```python
GalGenConfig(name='variant1', stamp_size=50, pixel_scale=.2, flux=1.)
```
you can call it with:
```python
tfds.load('GalGen/variant1')
```

**Comment by EiffL** [comment]
\o/ this is great :-D Thanks @b-remy

**Review by EiffL — APPROVED** [review]
LGTM!

**Comment by EiffL** [comment]
Many thanks @andrevitorelli  and @b-remy ! Let's use this dataset as our baseline for now, we can add additional features as they are needed for the next steps of the project, for instance PSF and/or Fourier image of the galaxies. 

I'm going to go ahead and merge this :-)

**Pull request closed without merging**